libfwupdplugin/fu-udev-device.c

```diff
@@ -1259,23 +1259,62 @@ fu_udev_device_match_subsystem_devtype(GUdevDevice *udev_device,
 	}
 	return TRUE;
 }
+#endif
 
-static GUdevDevice *
-fu_udev_device_get_parent_with_subsystem_devtype(GUdevDevice *udev_device,
+/**
+ * fu_udev_device_get_parent_with_subsystem_devtype
+ * @self: a #FuUdevDevice
+ * @subsystem: (nullable): the name of a udev subsystem
+ * @devtype: (nullable): the name of a udev devtype, e.g. `usb_interface`
+ * @error: (nullable): optional return location for an error
+ *
+ * Get the device that is a ancestor of self and has the provided subsystem and device type.
+ *
+ * Returns: (transfer full): device, or %NULL
+ *
+ * Since: 2.0.0
+ */
+FuUdevDevice *
+fu_udev_device_get_parent_with_subsystem_devtype(FuUdevDevice *self,
```

Comment:
sorry I should have made my comment clearer.  Not just the comment needs updating, I was suggesting the function too.

IE `fu_udev_device_get_ancestor_with_subsystem_devtype`

Reply:
I'm not sure ancestor works -- it makes it sound like it's an old device that's no longer here. Counterproposal -- we keep the existing `fu_udev_device_get_parent_with_subsystem()` and just add the `devtype` to that-- with `NULL` being "don't care"?

Reply:
narrator: he did that, and liked it.